### PR TITLE
Ian/airgap

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Change description
* Air gapped documentation added to governance notebook.
* Increased functionality of CredoGovernance object. It now can:
   * Handle registration of models and projects
   * Handle retrieving spec from governance platform
   * Handle retrieving spec from file
* Documentation for explicit registration of models updated to reflect use of credogovernance
* Better warning, exception and verbose functionality for Lens
* dataset registration

### CODE REVIEW

File: credoai/lens.py [hunk 1]
```diff
@@ -44,15 +44,87 @@ class CredoGovernance:
...
+        """
+        if not self.assessment_spec:
+            self.retrieve_assessment_spec()
+        spec = {}
+        metrics = self.assessment_spec
+        if metrics:           
+            spec['metrics'] = list(metrics.keys())
+        return {"FairnessBase": spec}
+
+    def get_info(self):
```
Comment (amrasekh) [inline credoai/lens.py:79]: Would be helpful to add docstring to this function

Comment (IanAtCredo) [reply credoai/lens.py:102]: Done

**Comment by IanAtCredo** [comment]
@amrasekh I noticed you use logging from absl in nlpgenerator. I created some not-good verbose printing. I should switch to your logging I think. Let's talk about it during sync tomorrow.

**Comment by amrasekh** [comment]
A couple of issue in [connection demo](https://github.com/credo-ai/credoai_lens/blob/ian/airgap/docs/notebooks/governance_integration.ipynb) 

- The following cell throws error. It seems the parameter should be `model_project_id`.
```
gov = cl.CredoGovernance(project_id = 'existing project')
gov.register_model(model_name='my_model')
```
- This last cell created the new project on the platform al right (I named it `my_project amin 2`), but it is empty (the model is not registered)

**Comment by IanAtCredo** [comment]
> @amrasekh I noticed you use logging from absl in nlpgenerator. I created some not-good verbose printing. I should switch to your logging I think. Let's talk about it during sync tomorrow.

Moved to just using logging. You can change which things are reported using `logging.set_verbosity`, though that isn't communicated anywhere. Hopefully someone who cares can find that we use absl

**Review by amrasekh — APPROVED** [review]

**Pull request merged**